src/main/java/org/wildfly/extension/elytron/CredentialStoreAliasDefinition.java

```diff
@@ -0,0 +1,258 @@
+import org.wildfly.security.credential.store.UnsupportedCredentialTypeException;
+import org.wildfly.security.password.PasswordFactory;
+import org.wildfly.security.password.interfaces.ClearPassword;
+import org.wildfly.security.password.spec.ClearPasswordSpec;
+
+/**
+ * A {@link ResourceDefinition} for an entry stored in credential store.
+ *
+ * @author <a href="mailto:pskopek@redhat.com">Peter Skopek</a>
+ */
+class CredentialStoreAliasDefinition extends SimpleResourceDefinition {
+
+    static final String OTHER = "Other";
+    private static final Class<?>[] SUPPORTED_CREDENTIAL_TYPES = {
+            PasswordCredential.class
+    };
+
+    static final SimpleAttributeDefinition ENTRY_TYPE;
+
+    static {
+        List<String> entryTypes = Stream.of(SUPPORTED_CREDENTIAL_TYPES).map(Class::getName)
+                .collect(Collectors.toList());
+        entryTypes.add(OTHER);
+        ENTRY_TYPE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ENTRY_TYPE, ModelType.STRING)
+                .setStorageRuntime()
+                .setAllowNull(true)
+                .setAllowedValues(entryTypes.toArray(new String[entryTypes.size()]))
+                .build();
+    }
+
+    static final StandardResourceDescriptionResolver RESOURCE_DESCRIPTION_RESOLVER = ElytronExtension.getResourceDescriptionResolver(ElytronDescriptionConstants.CREDENTIAL_STORE, ElytronDescriptionConstants.ALIAS);
+
+    static final SimpleAttributeDefinition SECRET_VALUE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.SECRET_VALUE, ModelType.STRING)
+            .setStorageRuntime()
+            .setAllowNull(false)
+            .build();
+
+    private static final AttributeDefinition[] CONFIG_ATTRIBUTES = new AttributeDefinition[] {SECRET_VALUE, ENTRY_TYPE};
+    //private static final WriteAttributeHandler WRITE = new WriteAttributeHandler();
+
+    private static final SimpleOperationDefinition ADD_DEFINITION = new SimpleOperationDefinitionBuilder(ModelDescriptionConstants.ADD,
+            RESOURCE_DESCRIPTION_RESOLVER)
+            .setParameters(SECRET_VALUE, ENTRY_TYPE)
+            .build();
+    private static final AddHandler ADD_HANDLER = new AddHandler();
+
+    CredentialStoreAliasDefinition() {
+        super(new Parameters(PathElement.pathElement(ElytronDescriptionConstants.ALIAS), RESOURCE_DESCRIPTION_RESOLVER)
+                .setRemoveHandler(new RemoveHandler())
+                .setAddRestartLevel(OperationEntry.Flag.RESTART_NONE)
+                .setRemoveRestartLevel(OperationEntry.Flag.RESTART_NONE)
+                .setRuntime()
+        );
+    }
+
+    @Override
+    public void registerOperations(ManagementResourceRegistration resourceRegistration) {
+        super.registerOperations(resourceRegistration);
+        // We needed a custom add operation so we could specify the parameters.
+        resourceRegistration.registerOperationHandler(ADD_DEFINITION, ADD_HANDLER);
```

Comment:
As the ADD handler is static we should be able to add it to the paramers in the constructor.